androidApp/src/main/java/com/example/tasky/android/login/screen/LoginScreen.kt

```diff
@@ -103,27 +108,50 @@ private fun LoginScreen(
             horizontalAlignment = Alignment.CenterHorizontally,
         ) {
             Spacer(Modifier.height(50.dp))
-            LoginTextField(
-                text = state.email,
+            CheckTextField(
+                state = state.emailState,
                 onTextChange = {
-                    onEvent(LoginScreenEvent.OnStateChange(state.copy(email = it)))
+                    onEvent(
+                        LoginScreenEvent.OnStateChange(
+                            state.copy(
+                                emailState =
+                                    state.emailState.copy(
+                                        text = it,
+                                    ),
+                            ),
+                        ),
```

Comment:
I think I mentioned this in a previous PR already, but it's the responsibility of the ViewModel to decide what happens with a UI event. The UI should not decide how state gets updated, as that will (1) mix responsibilities and (2) make it a lot harder to properly test this code since you'd need to run actual slow UI tests to test simple state update logic. 
Instead, I recommend to just let the VM know about the new text and the ViewModel then decides which part of the state gets updated with this new text (the `copy()` is done by the VM)